### PR TITLE
feat: add "pause" method to all connectors

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

- we need to be able to pause/resume connectors for https://github.com/dust-tt/tasks/issues/599
- we currently have stop/resume methods, but those are ill-defined and not implemented on all connectors. They are mostly used to restart connectors that rely on long-running workflows
- Kind of in a rush, because https://github.com/dust-tt/tasks/issues/599 is currently causing a lot of operational toil
- short term, we want to add the "unpause" method. 
- mid term, we want to cleanup the "connectors" interface, likely w/ an abstract base class. We likely will rename / delete / create some methods.

## Risk

not much, purely additive

## Deploy Plan

- run initdb for connectors on prodbox
- deploy code

### CODE REVIEW

File: connectors/src/api/webhooks/webhook_intercom.ts [hunk 1]
```diff
@@ -101,8 +102,18 @@ const _webhookIntercomAPIHandler = async (
     return res.status(200).end();
   }
 
-  // Check we have the permissions to sync this conversation
+  if (connector?.pausedAt) {
```
Comment (flvndvd) [inline connectors/src/api/webhooks/webhook_intercom.ts:105]: Given that we check on line 95.
```suggestion
  if (connector.pausedAt) {
```

File: connectors/src/connectors/github/index.ts [hunk 2]
```diff
@@ -142,6 +142,18 @@ export async function stopGithubConnector(
...
 
+export async function pauseGithubWebhooks(
+  connectorId: ModelId
+): Promise<Result<undefined, Error>> {
+  const connector = await ConnectorResource.fetchById(connectorId);
+  if (!connector) {
+    logger.error({ connectorId }, "Connector not found");
+    return new Err(new Error("Connector not found"));
+  }
+  await connector.update({ pausedAt: new Date() });
```
Comment (flvndvd) [inline connectors/src/connectors/github/index.ts:153]: Let's create a method on `ConnectorResource`. Something like `markAsPaused`.

File: connectors/src/connectors/google_drive/index.ts [hunk 3]
```diff
@@ -825,3 +825,12 @@ export async function googleDriveGarbageCollect(connectorId: ModelId) {
 
   return launchGoogleGarbageCollector(connectorId);
 }
+
+export async function pauseGoogleDriveWebhooks(connectorId: ModelId) {
```
Comment (flvndvd) [inline connectors/src/connectors/google_drive/index.ts:829]: Any reason to have Webhooks in the name of the function? I'd be in favor of using generic names, like for other connectors.

File: connectors/src/resources/connector_resource.ts [hunk 4]
```diff
@@ -114,6 +114,18 @@ export class ConnectorResource extends BaseResource<ConnectorModel> {
     return new this(this.model, blob.get());
   }
 
+  static async fetchByIds(ids: number[]) {
```
Comment (flvndvd) [inline connectors/src/resources/connector_resource.ts:117]:
```suggestion
  static async fetchByIds(ids: ModelId[]) {
```

File: connectors/src/resources/storage/models/connector_model.ts [hunk 5]
```diff
@@ -105,10 +107,19 @@ ConnectorModel.init(
...
+    },
   },
   {
     sequelize: sequelizeConnection,
     modelName: "connectors",
-    indexes: [{ fields: ["workspaceId", "dataSourceName"], unique: true }],
+    indexes: [
+      { fields: ["workspaceId", "dataSourceName"], unique: true },
+      {
+        fields: ["pausedAt"],
```
Comment (flvndvd) [inline connectors/src/resources/storage/models/connector_model.ts:121]: Any reason to add this index?

File: connectors/src/connectors/intercom/index.ts [hunk 6]
```diff
@@ -740,3 +740,19 @@ export async function retrieveIntercomContentNodeParents(
 
   return new Ok(parents);
 }
+
```
Comment (PopDaph) [inline connectors/src/connectors/intercom/index.ts:743]: This looks good and should pause the cron sync of the Help Center. 
Should we also pause the webhook processing for conversations here? https://github.com/dust-tt/dust/blob/main/connectors/src/api/webhooks/webhook_intercom.ts#L91-L123

Comment (PopDaph) [reply connectors/src/connectors/intercom/index.ts:743]: You've done that already!! 🤦🏻‍♀️  Closing this 😬

File: connectors/src/resources/storage/models/connector_model.ts (hunk 5 above)
Comment (fontanierh) [reply connectors/src/resources/storage/models/connector_model.ts:121]: I was expecting to query by pausedAt = null but turns out I am not

**Review by PopDaph — APPROVED** [review]

**Review by lasryaric — COMMENTED** [review]
`terminateAllWorkflowsForConnectorId()` logic discussed on Slack looks good to me 🙏

**Review by flvndvd — APPROVED** [review]
LGTM

**Pull request merged**